src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -45,33 +46,48 @@ class DashboardWidgetQuerySerializer(CamelSnakeSerializer):
     fields = serializers.ListField(child=serializers.CharField(), required=False)
     name = serializers.CharField(required=False, allow_blank=True)
     conditions = serializers.CharField(required=False, allow_blank=True)
+    orderby = serializers.CharField(required=False, allow_blank=True)
 
     required_for_create = {"fields", "conditions"}
 
     validate_id = validate_id
 
-    def validate_fields(self, fields):
-        snuba_filter = get_filter("")
+    def validate(self, data):
+        if not data.get("id"):
+            keys = set(data.keys())
+            if keys.intersection(self.required_for_create) != self.required_for_create:
+                raise serializers.ValidationError(
+                    {
+                        "fields": "fields are required during creation.",
+                        "conditions": "conditions are required during creation.",
+                    }
+                )
+
+        # Validate the query that would be created when run.
+        conditions = self._get_attr(data, "conditions", "")
+        fields = self._get_attr(data, "fields", [])
+        orderby = self._get_attr(data, "orderby", "")
         try:
-            resolve_field_list(fields, snuba_filter)
-            return fields
+            snuba_filter = get_filter(conditions)
         except InvalidSearchQuery as err:
-            raise serializers.ValidationError("Invalid fields: {}".format(err))
+            raise serializers.ValidationError({"conditions": f"Invalid conditions: {err}"})
 
-    def validate_conditions(self, conditions):
+        if orderby:
+            snuba_filter.orderby = get_function_alias(orderby)
         try:
-            get_filter(conditions)
+            resolve_field_list(fields, snuba_filter)
         except InvalidSearchQuery as err:
-            raise serializers.ValidationError("Invalid conditions: {}".format(err))
-        return conditions
-
-    def validate(self, data):
-        if not data.get("id"):
-            keys = set(data.keys())
-            if keys.intersection(self.required_for_create) != self.required_for_create:
-                raise serializers.ValidationError("The fields and conditions fields are required")
+            raise serializers.ValidationError({"fields": f"Invalid fields: {err}"})
         return data
 
+    def _get_attr(self, data, attr, empty_value=None):
+        value = data.get(attr)
+        if value:
```

Comment:
Should this be an explicit `if value is not None`? Take `conditions` for example, I believe the falsy value `""` is valid.